packages/components/src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,106 @@
+import type { SystemProp, Theme } from "@xstyled/styled-components";
+import React from "react";
+import * as HeroOutlineIcons from "@heroicons/react/24/outline";
+import { Box } from "../../primitives/Box";
+import { Icon } from "../Icon";
+
+type IconNames = keyof typeof HeroOutlineIcons;
+
+type BadgeSizeOptions = "large" | "small";
+type BadgeVariantOptions =
+  | "active"
+  | "inactive"
+  | "primary"
+  | "success"
+  | "warning";
+
+export interface BadgeProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, "color"> {
+  /** The contents of the badge. Can be text. */
+  children: NonNullable<React.ReactNode>;
+  /** Sets the badge to be disabled. This is still being discussed with design whether we need the disabled prop or not. */
+  isDisabled?: boolean;
+  /** Icon to be added on the left of the content. */
+  icon?: IconNames;
+  /** Sets the size of the badge. */
+  size?: BadgeSizeOptions;
+  /** Sets the variant of the badge. */
+  variant?: BadgeVariantOptions;
+}
+
+const getVariantProps = (
+  variant: BadgeVariantOptions
+): {
+  backgroundColor?: SystemProp<keyof Theme["colors"], Theme>;
+  color?: SystemProp<keyof Theme["colors"], Theme>;
+} => {
+  switch (variant) {
+    case "active": {
+      return {
+        backgroundColor: "colorBackgroundPrimary",
+        color: "colorTextInverse",
+      };
+    }
+    case "inactive": {
+      return {
+        backgroundColor: "colorBackgroundWeak",
+        color: "colorText",
+      };
+    }
+    case "success": {
+      return {
+        backgroundColor: "colorBackgroundSuccess",
+        color: "colorTextSuccess",
+      };
+    }
+    case "warning": {
+      return {
+        backgroundColor: "colorBackgroundWarning",
+        color: "colorTextWarning",
+      };
+    }
+    default: {
+      return {
+        backgroundColor: "colorBackgroundInfo",
+        color: "colorTextInfo",
+      };
+    }
+  }
+};
+
+const getIcon = (iconName: IconNames, size: BadgeSizeOptions) => {
+  const iconProps = (size: BadgeSizeOptions) =>
+    size === "small" ? "sizeIcon30" : "sizeIcon40";
+
+  return <Icon decorative icon={iconName} size={iconProps(size)} />;
```

Comment:
Could an icon be on the right side? :thinking:  Or is it always on the left (maybe it would be better to check it with the design team)?

Reply:
It seems it's always on the left side. https://localyzeteam.slack.com/archives/C03DUMP3XD4/p1666624596425449